libevmasm/Assembly.cpp

```diff
@@ -1359,35 +1394,59 @@ LinkerObject const& Assembly::assembleLegacy() const
 			{
 				if (i != offsets.size() - 1)
 				{
-					ret.bytecode.push_back(uint8_t(Instruction::DUP2));
-					ret.bytecode.push_back(uint8_t(Instruction::DUP2));
+					assembleInstruction([&]() {
+						ret.bytecode.push_back(uint8_t(Instruction::DUP2));
+					});
+					assembleInstruction([&]() {
+						ret.bytecode.push_back(uint8_t(Instruction::DUP2));
+					});
 				}
-				// TODO: should we make use of the constant optimizer methods for pushing the offsets?
-				bytes offsetBytes = toCompactBigEndian(u256(offsets[i]));
-				ret.bytecode.push_back(static_cast<uint8_t>(pushInstruction(static_cast<unsigned>(offsetBytes.size()))));
-				ret.bytecode += offsetBytes;
-				ret.bytecode.push_back(uint8_t(Instruction::ADD));
-				ret.bytecode.push_back(uint8_t(Instruction::MSTORE));
+				assembleInstruction([&]() {
+					// TODO: should we make use of the constant optimizer methods for pushing the offsets?
+					bytes offsetBytes = toCompactBigEndian(u256(offsets[i]));
+					ret.bytecode.push_back(static_cast<uint8_t>(pushInstruction(static_cast<unsigned>(offsetBytes.size()))));
+					ret.bytecode += offsetBytes;
+				});
+				assembleInstruction([&]() {
+					ret.bytecode.push_back(uint8_t(Instruction::ADD));
+				});
+				assembleInstruction([&]() {
+					ret.bytecode.push_back(uint8_t(Instruction::MSTORE));
+				});
 			}
 			if (offsets.empty())
 			{
-				ret.bytecode.push_back(uint8_t(Instruction::POP));
-				ret.bytecode.push_back(uint8_t(Instruction::POP));
+				assembleInstruction([&]() {
+					ret.bytecode.push_back(uint8_t(Instruction::POP));
+				});
+				assembleInstruction([&]() {
+					ret.bytecode.push_back(uint8_t(Instruction::POP));
+				});
 			}
 			immutableReferencesBySub.erase(item.data());
 			break;
 		}
 		case PushDeployTimeAddress:
-			ret.bytecode += assemblePushDeployTimeAddress();
+			assembleInstruction([&]() {
+				ret.bytecode += assemblePushDeployTimeAddress();
+			});
 			break;
 		case Tag:
-			ret.bytecode += assembleTag(item, ret.bytecode.size(), true);
+			assembleInstruction([&](){
+				ret.bytecode += assembleTag(item, ret.bytecode.size(), true);
+			});
 			break;
 		default:
 			solAssert(false, "Unexpected opcode while assembling.");
 		}
+
+		++assemblyItemIndex;
 	}
 
+	codeSectionLocation.end = ret.bytecode.size();
+
+	ret.codeSectionLocations.emplace_back(codeSectionLocation);
```

Comment:
```suggestion
	ret.codeSectionLocations.emplace_back(std::move(codeSectionLocation));
```
Again, functionally equivalent, but it would save some cycles (especially since we don't always need this). But I'd also not let this block merging.